src/main/daml/Daml/Finance/Interface/Instrument/Swap/CreditDefault.daml

```diff
@@ -0,0 +1,82 @@
+import Daml.Finance.Interface.Util.Disclosure qualified as Disclosure (I, Implementation)
+
+-- | Type synonym for `Factory`.
+type F = Factory
+
+-- | Type synonym for `View`.
+type V = View
+
+-- | View of `Factory`.
+data View = View
+  with
+    provider : Party
+      -- ^ The provider of the `Factory`.
+  deriving (Eq, Ord, Show)
+
+-- | Interface that allows implementing templates to create credit default swaps.
+interface Factory where
+  viewtype V
+
+  asDisclosure : Disclosure.I
+    -- ^ Conversion to `Disclosure` interface.
+  create' : Create -> Update (ContractId Instrument.I)
+    -- ^ Implementation of `Create` choice.
+  remove : Remove -> Update ()
+    -- ^ Implementation of `Remove` choice.
+
+  nonconsuming choice Create : ContractId Instrument.I
+    -- ^ Create a new instrument.
+    with
+      instrument : InstrumentKey
+        -- ^ The instrument's key.
+      description : Text
+        -- ^ The description of the swap.
+      defaultProbabilityReferenceId : Text
+        -- ^ The reference ID of the default probability observable. For example, in case of protection against a "TSLA a bond payment default" this should be a valid reference to the "TSLA default probability".
+      recoveryRateReferenceId : Text
+        -- ^ The reference ID of the recovery rate observable. For example, in case of a "TSLA a bond payment default with a 60% recovery rate" this should be a valid reference to the "TSLA bond recovery rate".
+      issuerPaysFix : Bool
+        -- ^ Indicate whether the issuer of the instrument pays the fix rate leg or the default protection leg of the swap.
+      fixRate : Decimal
+        -- ^ The interest rate of the fix leg. For example, in case of "2.5% fix" this should be 0.025.
+      periodicSchedule : PeriodicSchedule
+        -- ^ The schedule for the periodic swap payments.
+      holidayCalendarIds : [Text]
+        -- ^ The identifier of the holiday calendar to be used for the swap payment schedule.
+      calendarDataProvider : Party
+        -- ^ The reference data provider to use for the holiday calendar.
+      dayCountConvention : DayCountConventionEnum
+        -- ^ The day count convention used to calculate day count fractions. For example: Act360.
+      currency : Instrument.K
+        -- ^ The currency of the swap. For example, if the swap pays in USD this should be a USD cash instrument.
+      lastEventTimestamp : Time
+        -- ^ (Market) time of the last recorded lifecycle event. If no event has occurred yet, the time of creation should be used.
+      observers : PartiesMap
+        -- ^ The instrument's observers.
+    controller instrument.depository, instrument.issuer
+    do
+      create' this arg
+
+  nonconsuming choice Remove : ()
+    -- ^ Archive an instrument.
+    with
+      instrument : InstrumentKey
+        -- ^ The instrument's key.
+    controller instrument.depository, instrument.issuer
+      do
+        remove this arg
```

Comment:
```suggestion
```
This is not needed as with the authorization of depository and issuer we can just plainly `archive` the contract.

Reply:
Ah sorry, I didn't realize this was on the factory. Removing this suggestion